lib/cli.nix

```
@@ -90,15 +90,33 @@ rec {
     mkOption ?
       k: v: if v == null
             then []
-            else [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ]
+            else [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ],
```

The current implementation looks rather backwards. Shouldn't something simple like

```suggestion
            else if optionValueSeparator == null then
              [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ]
            else
              [ "${mkOptionName k}${optionValueSeparator}${lib.generators.mkValueStringDefault {} v}" ],
```

work?

---

Thank you, yes that's much better